activestorage/lib/active_storage/service/disk_service.rb

```diff
@@ -122,18 +132,8 @@ def private_url(key, expires_in:, filename:, content_type:, disposition:, **)
           port: current_uri.port,
           disposition: content_disposition,
           content_type: content_type,
-          filename: filename
-        )
-      end
-
-      def public_url(key, filename:, content_type: nil, disposition: :attachment, **)
-        current_uri = URI.parse(current_host)
-
-        url_helpers.rails_disk_service_public_url(key,
-          protocol: current_uri.scheme,
-          host: current_uri.host,
-          port: current_uri.port,
-          filename: filename
+          filename: filename,
+          service_name: name
```

Comment:
I don't think we need this query param, do we?

Reply:
I included it to be consistent with how the `disposition` and `content_type` attributes are included. I think that the `disposition` and `content_type` query parameters are not used either since it's included in the verified key. Looking at the [git history of this file](https://github.com/rails/rails/blob/8f125d5b7970727064895f67887a59c1cedcf273/lib/active_storage/service/disk_service.rb#L39), I suspect this remains because, originally, the verified key only encoded the `key` and not the disposition nor the content type. AFAIK the disk controller doesn't use these query parameters either. I think it's safe to remove these query params unless there's a use case that I'm not aware of. WDYT?

Reply:
I created a PR that removes these query parameters, please share your thoughts at #37904!

Reply:
Seems fine to remove the old params in that PR. Can you remove `service_name` as a param in this PR?

Reply:
Done!